tests/test_synthetic_pii.py

```diff
@@ -731,3 +732,72 @@ def test_id_uniqueness(artifact, hdf_key, num_need_ids, random_seed):
         pytest.mark.skip(reason=f"Cannot find artifact at {artifact.path}")
     else:
         assert len(np.unique(ids)) == num_need_ids
+
+
+def test_copy_household_member():
+    # Dummy dataset
+    # Note this function requires columns "household_id" and will copy from "age", "date_of_birth", and
+    # "ssn" if that column is present
+    hh_ids = [0, 0, 21, 21, 21, 21, 12, 12, 13, 13, 14, 14, 14, 14, 14, 25, 25, 25, 26, 27]
+    ages = [
+        10,
+        11,
+        12,
+        13,
+        14,
+        15,
+        16,
+        17,
+        18,
+        19,
+        20,
+        21,
+        22,
+        23,
+        24,
+        25,
+        26,
+        27,
+        28,
+        29,
+    ]
+    dob = ["1976-07-22", "1992-04-07", "1996-05-08", "2001-04-08", "1999-05-23"] * 4
+    ids = list(range(1000, 1020))
+    has_ssn = [True, False] * 10
+
+    pop = pd.DataFrame(
+        {
+            "household_id": hh_ids,
+            "age": ages,
+            "ssn": ids,
+            "date_of_birth": dob,
+            "has_ssn": has_ssn,
+        },
+        index=list(range(2000, 2020)),
+    )
+
+    copy = copy_from_household_member(pop, randomness)
+
+    copy_cols = metadata.COPY_HOUSEHOLD_MEMBER_COLS
+    for col in [column for column in copy_cols.keys() if column != "has_ssn"]:
+        # Check expected missing rows based on household for age and date of birth
+        no_copy_idx = copy.index[copy[copy_cols[col]].isna()]
+        # Get household ids that are in GQ or only have a single occupant
+        expected_no_copy_idx = pop.index[
+            pop["household_id"].isin(list(data_values.GQ_HOUSING_TYPE_MAP.keys()) + [26, 27])
+        ]
+        assert no_copy_idx.equals(expected_no_copy_idx)
+
+    # Check expected missing rows for ssns. Note: These will be the same household ids along
+    # with household ids 12, 13 and 25 because those households only have 1 member with a SSN
+    no_copy_idx = copy.index[copy["copy_ssn"].isna()]
+    expected_no_copy_idx = pop.index[
+        pop["household_id"].isin(
+            list(data_values.GQ_HOUSING_TYPE_MAP.keys()) + [12, 13, 25, 26, 27]
+        )
+    ]
+    assert no_copy_idx.equals(expected_no_copy_idx)
+    # SSNs are unique so we cannot copy the original value like we can with age and date of birth so
+    # we can check that there are no matches
+    have_ssn_copy = copy.index.difference(no_copy_idx)
+    assert (copy.loc[have_ssn_copy, "ssn"] != copy.loc[have_ssn_copy, "copy_ssn"]).all()
```

Comment:
Do you think it's worth adding a test to ensure you are uniformly choosing from available values?

Reply:
I think its a good idea but I'm not sure how useful or accurate we could make it since we have so many 2-3 person households.